database/gormdb.go

```diff
@@ -337,6 +337,7 @@ func (db *GormDB) GetNextAssignment(cid uint64, uid uint64, gid uint64) (*pb.Ass
 }
 
 // CreateSubmission creates a new submission record
+// or updates the last one for the given lab and student/group
```

Comment:
Remove TODO below as we talked about—let’s keep access control out of the database package.

Reply:
Done.

Reply:
This should read:
```
or updates the most recent submission, as defined by the provided submissionQuery.
The submissionQuery must always specify the assignment, and may specify the ID of
either an individual student or a group, but not both.
```
(or something along those lines.)

Thinking about this more deeply. Maybe this should be split into two, one `CreateSubmission` and another for `UpdateSubmission`.

I'm also wondering if it makes more sense to move the "preparation" of the query outside this (and other similar) functions, rather than allowing to pass the submission details... This would make the function work more like database query thing.

Reply:
PS: We already have a method called `UpdateSubmission`, which only touches the `approved` field. I suppose this method could be redesigned to support both updating the approved field and other submission fields.

Reply:
Related to my above point about preparing the query outside this method; this is also done in `GetSubmission` and `GetSubmissions`. But not in `CreateAssignment`. However, this is not a query method, so maybe it doesn't make sense to do it for that method... PS: The latter method does not seem to assign the "score limit" (not sure what assign does).

Reply:
> PS: The latter method does not seem to assign the "score limit" (not sure what assign does).

It does in master, was part of the previous PR. Will update this branch along with above-mentioned changes.

Reply:
Shall we keep this as is for this PR, and do another to split the Create and Update functionality and add query input support?

Reply:
Either is fine for me.